src/Language/Haskell/LSP/Core.hs

```diff
@@ -844,6 +844,10 @@ initializeRequestHandler' onStartup mHandler tvarCtx req@(J.RequestMessage _ ori
           static (Just d) _ = Just d
           static _ (Just _) = Just (J.GotoOptionsStatic True)
           static _ Nothing  = Nothing
+          
+          static' (Just d) (Just _) = Just d
+          static' _ (Just _) = Just (J.CodeActionOptionsStatic True)
+          static' _ _  = Nothing
```

Comment:
You could modify `static` to take an extra default argument, then you can use it for both the cases.

Reply:
I am happy with either option.

Reply:
Cases are not identical, if I see that correctly.
`static'` only returns `Just d` if both provider and handler are set, while `static` only requires the provider?

Reply:
Yeah, but why are they different?